uk.co.compucorp.civicrm.hrleaveandabsences/tests/phpunit/CRM/HRLeaveAndAbsences/BAO/LeavePeriodEntitlementTest.php

```diff
@@ -519,6 +511,44 @@ public function testSaveFromEntitlementCalculationCanSaveOverriddenValuesLessTha
     $this->assertEquals($contact['id'], $periodEntitlement->contact_id);
     $this->assertEquals(1, $periodEntitlement->overridden);
     $this->assertEquals($overriddenEntitlement, $periodEntitlement->getEntitlement());
+  }
+
+  public function testCanSaveALeavePeriodEntitlementWithAComment() {
+    $userId = 1;
+    $this->registerCurrentLoggedInContactInSession($userId);
+
+    $type = new AbsenceType();
+    $type->id = 1;
+    $period = new AbsencePeriod();
+    $period->id = 1;
+    $contact = ['id' => 2];
+
+    $broughtForward = 1;
+    $proRata = 10;
+    $overridden = false;
+    $calculation = $this->getEntitlementCalculationMock(
+      $period,
+      $contact,
+      $type,
+      $broughtForward,
+      $proRata,
+      [],
+      $overridden
+    );
+
+    $comment = 'Lorem ipsum dolor sit amet...';
+    LeavePeriodEntitlement::saveFromCalculation($calculation, null, $comment);
+
+    $periodEntitlement = LeavePeriodEntitlement::getPeriodEntitlementForContact(
+      $contact['id'],
+      $period->id,
+      $type->id
+    );
+
+    $this->assertNotNull($periodEntitlement);
+    $this->assertEquals($comment, $periodEntitlement->comment);
+    $this->assertNotNull($periodEntitlement->comment_date);
```

Comment:
Instead of just asserting that the comment_date is not NULL. You can do something like this [Here](https://github.com/civicrm/civihr/blob/d2b646d4d22e90ae978dc658d79929ec7df56b78/uk.co.compucorp.civicrm.hrleaveandabsences/tests/phpunit/CRM/HRLeaveAndAbsences/Mail/Template/SicknessRequestNotificationTemplateTest.php#L85-L85). I.e you compare the comment date to the current time and check that the delta is within a certain range(in the link example, the delta is 10 secs). Both time will be in timestamp when comparing.